app.rb

```diff
@@ -1,20 +1,136 @@
+    # The propreties for the apartments hash type are: ID, address, monthly_rent, and square_feet. Properties for the tenants has type are:
+    # name, age, and apartment_id.
 
 # Use enumerables to -
   # Print all the addresses for the apartments
+  # apartments.each do |apartment|
+  #   puts apartment[:address]
+  # end
+
   # Print all the names for tenants
+  # tenants.each do |tenant|
+  #   puts tenant[:name]
+  # end
+
   # Print only apartments that are less then 700 in rent
+  # cheap_apartments = apartments.select do |apartment|
+  #   apartment[:monthly_rent] < 700
+  # end
+  # puts cheap_apartments
+
   # Print only tenants that are over the age of 44
+  # old_tenants = tenants.select do |tenant|
+  #   tenant[:age] > 44
+  # end
+  # puts old_tenants
+
   # Print only tenants that have an apartment id of 1
+    # tenants_id = tenants.select do |tenant|
+    #   tenant[:apartment_id] == 1
+    # end
+    # puts tenants_id
+
   # Print all the tenants in order from youngest to oldest
+    # tenants_order_by_age = tenants.sort_by do |tenant|
+    #   tenant[:age]
+    # end
+    # puts tenants_order_by_age
+
   # Print the names of all the tenants alphabetically
+    # tenants_alpha = tenants.sort_by do |tenant|
+    #   tenant[:name].downcase
+    # end
+    # puts tenants_alpha
 
   ## More challenging
-  # When printing tenants also print out the address that the tenant resides in.
   # When printing all apartments, under each apartment print all of its tenants
+
+
+  puts "What is your name?"
+  name = gets.chomp
+  puts "OK I recognize that name. Welcome back " + name + "!"
+  puts "Please choose one of the following options:"
+    puts "1. Overview of all apartments"
+    puts "2. Overiview of all tenants"
+    puts "3. Overview of apartment occupancy"
+    puts "4. Please enter tenant "
+
+  choice = nil
+
+  while ![1, 2, 3, 4].include? choice
+    choice = (gets.chomp).to_i
+  end
+
+case choice
+  when 1
+  puts "All Apartments"
+  buffer
+    ruby_apartments.each do |apartment|
+      puts apartment.address
+    end
+
+  when 2
+    puts "All Tenants"
+    buffer
+      ruby_tenants.each do |tenant|
+        puts tenant.name
+      end
+
+  when 3
+      puts "Apartment Occupancies"
+      buffer
+      ruby_apartments.each do |apartment|
+        puts apartment.address
+        puts "Occupants:"
+        apartment_occupants = ruby_tenants.select do |tenant|
+          tenant.apartment_id == apartment.id
+        end
+        apartment_occupants.each do |tenant|
+          puts tenant.name
+      end
+    end
+
+  when 4
+    puts "Tenants name with Address"
+    buffer
+    ruby_tenants.each do |tenant|
+      puts tenant.name
+      puts "Address:"
+      tenant_address = ruby_apartments.select do |apartment|
+        apartment.id == tenant.apartment_id
+      end
+      tenant_address.each do |tenant|
+        puts tenant.address
+        buffer
+    end
+  end
```

Comment:
What you have for 4 is great, but it's not consistent with what you said in menu for 4/the prompt. You could do another `gets.chomp` here, store it in a variable, and then use that to select from tenants.